### DISCUSSION TITLE
How to add the image2 as an Additional plugins in this package?

### DISCUSSION THREAD

**Discussion by divya-kanak** [waynestate/nova-ckeditor4-field#80] [Q&A]
Hello!

I'm working with Nova 4 and integrated this package as an editor in one of my post modules.

Now, I need to integrate the below upload image plugin in CKeditor +1

https://ckeditor.com/cke4/addon/image2

I've followed this solution but didn't get the upload option in image pop up

https://github.com/waynestate/nova-ckeditor4-field/wiki/Adding-additional-plugins

can anyone please guide me on this?

@chrispelzer @robertvrabel @f-liva @dsampaolo @shaffe-fr @harrymessenger

**Comment by shaffe-fr** [comment]
Hi @divya-kanak,
To add the image2 plugin, I added the following to the `nova/ckeditor.php` config.
Make sure to use the `full-all` version in the `ckeditor_url` config.

```php
<?php

return [
    /*
    |--------------------------------------------------------------------------------
    | CKEditor Options
    |--------------------------------------------------------------------------------
    |
    | To view a list of all available options checkout the CKEditor API documentation
    | https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html
    |
    */
    'options' => [
        // 'contentsCss' => env('APP_URL', 'http://localhost').'/css/admin.css',

        'extraPlugins' => 'image2',
        //...
    ],
    //...
    'ckeditor_url' => 'https://cdn.ckeditor.com/4.20.0/full-all/ckeditor.js',
];
```

Here is my toolbar:
![image](https://user-images.githubusercontent.com/3834222/220669090-f218cef2-a6c7-409b-bcd1-250c05f2dd6b.png)

Karel

**Comment by chrispelzer** [comment]
Hi @divya-kanak, I took a deeper look into all this today and I did find an issue with the way that image2 was not being loaded if no user configuration (never published the configuration and set up your own extraplugins) which is now fixed in #82 with [v1.2.1](https://github.com/waynestate/nova-ckeditor4-field/releases/tag/1.2.1)

As for the Upload button, I did look into that and as of right now the only supported option for uploading is by Drag and Dropping since the [filebrowserUploadURL](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-filebrowserUploadUrl) isn't configured.

Though it seems that there are some [fallbacks](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_file_upload.html#example-setting-upload-url-for-uploading-pasted-and-dragged-content) in place, where if one isn't set then it uses the other, but I think the current set is that the final chain of the process. I'll look a little more into this.
